src/pages/requestResetPassword.tsx

```diff
@@ -0,0 +1,95 @@
+import { useContext, useState } from "react";
+import { NextRouter, useRouter } from "next/router";
+import { Box, Button, Card, Stack, Typography } from "@mui/material";
+import { grey } from "@mui/material/colors";
+
+import { useSupabaseClient } from "@supabase/auth-helpers-react";
+import { Database } from "supabase/db_types";
+import { userContext } from "supabase/user_context";
+
+import Background from "@/components/public/Background";
+import EmailTextField from "@/components/requestResetPassword/EmailTextField";
+import { validateEmail } from "@/utilities/validation";
+
+import { validation } from "@/types/Validation";
+import { PagePaths } from "enum/pages";
+import Loading from "@/components/public/Loading";
+import { RequestResetPassword } from "@/services/Password";
+
+export default function Home() {
+  const router: NextRouter = useRouter();
+  const supabaseClient = useSupabaseClient<Database>();
+  const userStatus = useContext(userContext);
+
+  const [email, setEmail] = useState<string>("");
+  const [isSubmit, setIsSubmit] = useState<boolean>(false);
+
+  const isEmailErr: validation = validateEmail(email);
+
+  function handleEmailChange(
+    event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
+  ): void {
+    setIsSubmit(false);
+    setEmail(event.target.value);
+  }
+
+  async function handleSubmit() {
+    setIsSubmit(true);
+
+    if (!isEmailErr.err) {
+      // reset password end point goes here
+      RequestResetPassword(email, supabaseClient).then(() => {
+        router.push(PagePaths.successRequestResetPassword)
+      }).catch((err) => {
+        setIsSubmit(false);
+        console.log(err)
+        return
+      })
+    }
+  }
+
+  if (userStatus.isLoading) return <Loading />
+  return (
+    <>
+      {isSubmit && <Loading />}
```

Comment:
`Loading` really took so long. After press button, if the error occurs, the error message didn't took so long to appear unlikely the loading which still appear so long that I have to reload the page. Otherwise, loading took not so long before `successRequestResetPassword` page appears (but hopefully it could be faster).
The picture below is an example of what I mean.
<img width="1530" alt="Screenshot 2566-02-25 at 14 10 08" src="https://user-images.githubusercontent.com/62459518/221344469-148b77e7-3d66-42bd-abd0-b18be51b7c0a.png">

Reply:
This is @Jinnapat 's section. Let him elaborate his code.

Reply:
I misunderstood the use of isSubmit state. I'll use a new state to store whether are we waiting for backend response or not.